misc-scripts/xref_mapping/XrefParser/RefSeqCoordinateParser.pm

```diff
@@ -23,405 +23,302 @@ use strict;
-
-use base qw( XrefParser::BaseParser );
+use Readonly;
 use Bio::EnsEMBL::Registry;
 
-sub run_script {
+use parent qw( XrefParser::BaseParser );
+
+# Only scores higher than the threshold will be stored for both transcripts and translateable transcripts
+Readonly my $TRANSCRIPT_SCORE_THRESHOLD => 0.75;
+Readonly my $TL_TRANSCRIPT_SCORE_THRESHOLD => 0.75;
+# If Biotypes do not match, score will be multiplied with the penalty
+Readonly my $PENALTY => 0.9;
 
+sub run_script {
   my ($self, $ref_arg) = @_;
+
   my $source_id    = $ref_arg->{source_id};
   my $species_id   = $ref_arg->{species_id};
   my $species_name = $ref_arg->{species};
   my $file         = $ref_arg->{file};
-  my $verbose      = $ref_arg->{verbose};
   my $db           = $ref_arg->{dba};
-  my $dbi          = $ref_arg->{dbi};
-  $dbi = $self->dbi unless defined $dbi;
+  my $dbi          = $ref_arg->{dbi} // $self->dbi;
+  my $verbose      = $ref_arg->{verbose} // 0;
 
+  # initial param validation step
   if((!defined $source_id) or (!defined $species_id) or (!defined $file) ){
     croak "Need to pass source_id, species_id and file as pairs";
   }
-  $verbose |=0;
-
-  my $peptide_source_id = $self->get_source_id_for_source_name('RefSeq_peptide', 'otherfeatures', $dbi);
-  my $mrna_source_id = $self->get_source_id_for_source_name('RefSeq_mRNA', 'otherfeatures', $dbi);
-  my $ncrna_source_id = $self->get_source_id_for_source_name('RefSeq_ncRNA', 'otherfeatures', $dbi);
-
-  my $pred_peptide_source_id =
-    $self->get_source_id_for_source_name('RefSeq_peptide_predicted', 'otherfeatures', $dbi);
-  my $pred_mrna_source_id =
-    $self->get_source_id_for_source_name('RefSeq_mRNA_predicted','otherfeatures', $dbi);
-  my $pred_ncrna_source_id =
-    $self->get_source_id_for_source_name('RefSeq_ncRNA_predicted', 'otherfeatures', $dbi);
-
-  if($verbose){
-    print "RefSeq_peptide source ID = $peptide_source_id\n";
-    print "RefSeq_mRNA source ID = $mrna_source_id\n";
-    print "RefSeq_ncRNA source ID = $ncrna_source_id\n";
-    print "RefSeq_peptide_predicted source ID = $pred_peptide_source_id\n";
-    print "RefSeq_mRNA_predicted source ID = $pred_mrna_source_id\n" ;
-    print "RefSeq_ncRNA_predicted source ID = $pred_ncrna_source_id\n" ;
-  }
 
-  my $user = "ensro";
-  my $host;
-  my $port = 3306;
-  my $dbname;
-  my $pass;
-  my $transcript_score_threshold = 0.75;
-  my $tl_transcript_score_threshold = 0.75;
-  my $project;
-
-# Grep the project name, should be ensembl or ensemblgenomes
-  if($file =~ /project[=][>](\S+?)[,]/){
-    $project = $1;
-  }
+  my $file_params = $self->parse_file_string($file);
 
-# If specified, get core database connection details
-  if($file =~ /host[=][>](\S+?)[,]/){
-    $host = $1;
-  }
-  if($file =~ /port[=][>](\S+?)[,]/){
-    $port =  $1;
-  }
-  if($file =~ /dbname[=][>](\S+?)[,]/){
-    $dbname = $1;
+  # project or db param validation
+  if ( $file_params->{project} ne ('ensembl' || 'ensemblgenomes') && !defined $db ) {
```

Comment:
This doesn't really do what it is supposed to, does it? Correct me if I'm wrong but ( 'ensembl' || 'ensemblgenomes' ) should always evaluate to simply 'ensembl'.

Reply:
thanks for noticing. fixed